ibis/expr/datatypes/tests/test_core.py

```diff
@@ -518,6 +518,11 @@ def test_timestamp_unit():
         assert dt.Timestamp(scale=scale).unit == TimestampUnit.NANOSECOND
 
 
+def test_timestamp_unit_raise():
```

Comment:
I added this because it seemed like this https://github.com/ibis-project/ibis/blob/93552812ee9e8e0e3397bc226cc20c381fcd173b/ibis/expr/datatypes/core.py#L604-L606 wasn't covered. 

But turns out this test is causing a raise in a different place, which is confusing.

Reply:
The `Timestamp` op has a restriction that only the literal ints from 0 through 10 can be passed as `scale` -- so while it isn't technically unit-tested, it's not possible to set a value outside of that range without raising a `SignatureValidationError`

Reply:
That makes sense, but then why having a code path to "raise" that would never been triggered.

Reply:
i would wager that the `raise` predates the enforcement of values at the op level